rust/template/cmd_parser/parse.rs

```diff
@@ -563,6 +564,70 @@ fn test_struct() {
                                                                     (Cow::from("sfield"), Record::String("foo\nbar".to_string()))]))]))));
 }
 
+// nom's implementation mof `recognize_float` accepts integer literals
+// like `5`, making it impossible for the parser to distinguish between
+// floats and integers.  The following is a strict version of `recognize_float`
+// that only accepts literals containing `.`, e.g., `0.5` or `e|E`, e.g.,
+// `5e-1`.
+named!(recognize_float<&[u8], &[u8]>,
+  recognize!(
+    do_parse!(
+      opt!(alt!(char!('+') | char!('-'))) >>
+      alt!(
+          do_parse!(digit >>
+                    alt!( value!((),tuple!(char!('.'), opt!(digit), opt!(recognize_exponent)))
+                        | recognize_exponent) >>
+                    (()))
+        | do_parse!(char!('.') >>
+                   digit >>
+                   opt!(recognize_exponent) >>
+                   (()))) >>
+      (())
+    )
+  )
+);
+
+named!(recognize_exponent<&[u8], ()>,
+       do_parse!(alt!(char!('e') | char!('E')) >>
+                 opt!(alt!(char!('+') | char!('-'))) >>
+                 digit >>
+                 (()))
+);
+
+// We do not have enough type information to distinguish between floats
+// and doubles during parsing, so we parse all floating point numbers as
+// double's and postpone casting them to floats until when they get converted
```

Comment:
doubles, no apostrophe.
you may want a similar comment for bigint if it isn't there